app/views/download/help_improve.html.haml

```diff
@@ -16,41 +16,30 @@
 = content_for :javascript do
   :javascript
     (function(document, window, undefined) {
-      var forms = document.querySelectorAll('.js--ga-submit')
-
-      var getValue = function(element) {
-          if (element === null) {
-            return 'Blank'
-          }
-          else if (element.value) {
-            return element.value
-          }
-
-          return false;
+      var isIE = function() {
+        var myNav = navigator.userAgent.toLowerCase()
+        return (myNav.indexOf('msie') != -1) ? parseInt(myNav.split('msie')[1]) : false
       }
 
-      for (var i = 0; i < forms.length; i += 1) {
-        forms[i].addEventListener('submit', function(event) {
-          var form = event.target
+      if (isIE() && isIE() >= 9 || isIE() === false ) {
+        var forms = document.querySelectorAll('.js--ga-submit')
 
-          var isGovernment = getValue(form.querySelector('input[name="help_us_improve_user[is_government]"]:checked'))
-          var whatAreYouUsingRegistersFor = function() {
-            if (isGovernment === "Yes") {
-              return getValue(form.querySelector('input[name="help_us_improve_user[gov_what_are_you_using_registers_for]"]:checked'))
+        var sendData = function(form) {
+
+          var getWhatRegistersAreBeingUsedFor = function(form) {
```

Comment:
Since we've got a lot of code here to extract data from the form I'm wondering if it makes sense to have an object with all these functions as methods. I think it would help to separate out some of the different concerns (parsing the form, talking to GA, and presentational stuff).

Reply:
Agreed - I've done this in this [different branch](https://github.com/openregister/registers-frontend/pull/481), since a ticket was raised to make the entire form mandatory.